assets/js/comment-form.js

```diff
@@ -16,9 +16,14 @@ channel
   });
 
 channel.on('new-comment', payload => {
-  $(
-    `[data-announcement-id='${payload.announcement_id}'] .comments-list`
-  ).append(payload.comment_html);
+  $(`[data-announcement-id='${payload.announcement_id}'] .comments-list`)
+    .append(payload.comment_html);
+
+  if (payload.subscribe === true) {
+    const toggleSubscription = $('.subscription > a');
+    toggleSubscription.html('&nbsp;Subscribed to thread');
+    toggleSubscription.attr('data-method', 'delete');
+  }
```

Comment:
Prior to https://github.com/thoughtbot/constable/commit/bc4642e39b7e5253e99acf2f19cb25165d98a634 we were toggling different icons for subscribed/unsubscribed status on announcement threads via css classes. We'll have to rethink how to do that since https://github.com/thoughtbot/constable/pull/628/commits/caad791e177ba762d20dbf974a579c7ed9d3cecd no longer is viable.

Reply:
I opted for passing the entire button's html and replacing it altogether. It's a little strange extracting the button into its own partial so we can render it as a string from the channel, but it seems less brittle when someone changes text, css classes, etc. I updated it in https://github.com/thoughtbot/constable/pull/628/commits/6b1a92f9e3eff163c74b7ed058ce9a9f50697e76